common/lib/dependabot/shared_helpers.rb

```diff
@@ -75,13 +75,17 @@ def self.run_helper_subprocess(command:, function:, args:, env: nil,
       start = Time.now
       stdin_data = JSON.dump(function: function, args: args)
       cmd = allow_unsafe_shell_command ? command : escape_command(command)
-      env_cmd = [env, cmd].compact
+
+      # NOTE: For debugging native helpers in specs and dry-run: outputs the
+      # bash command to run in the tmp directory created by
+      # in_a_temporary_directory
       if ENV["DEBUG_FUNCTION"] == function
-        escaped_stdin_data = stdin_data.gsub("\"", "\\\"")
-        puts "$ cd #{Dir.pwd} && echo \"#{escaped_stdin_data}\" | #{env_cmd.join(' ')}"
+        puts helper_subprocess_bash_command(stdin_data: stdin_data, command: cmd, env: env)
```

Comment:
@brrygrdn moved this into it's own method

Example output, if you run `DEBUG_FUNCTION=parsed_gemspec bundle exec rspec spec/dependabot/bundler/file_parser_spec.rb:685` you'll see this in console:

```
$ cd /Users/feelepxyz/code/dependabot-core/bundler/tmp/dependabot_20210302-86366-1xt144c && echo "{\"function\":\"parsed_gemspec\",\"args\":{\"gemspec_name\":\"example.gemspec\",\"lockfile_name\":null,\"dir\":\"/Users/feelepxyz/code/dependabot-core/bundler/tmp/dependabot_20210302-86366-1xt144c\"}}" | BUNDLER_VERSION=1 BUNDLE_GEMFILE=/Users/feelepxyz/code/dependabot-core/bundler/lib/dependabot/bundler/../../../helpers/v1/Gemfile bundle exec ruby /Users/feelepxyz/code/dependabot-core/bundler/lib/dependabot/bundler/../../../helpers/v1/ru
```

Reply:
Nice, thanks for clarifying this 🙇🏻